lib/shared/src/sourcegraph-api/graphql/client.ts

```diff
@@ -515,16 +517,59 @@ export class SourcegraphGraphQLAPIClient {
         repositories: string[],
         query: string
     ): Promise<FuzzyFindFile[] | Error> {
-        return this.fetchSourcegraphAPI<APIResponse<FuzzyFindFilesResponse>>(FUZZY_FILES_QUERY, {
-            query: `type:path count:30 ${
-                repositories.length > 0 ? `repo:^(${repositories.map(escapeRegExp).join('|')})$` : ''
-            } ${query}`,
-        }).then(response =>
-            extractDataOrError(
-                response,
-                data => data.search?.results.results ?? new Error('no files found')
+        // as far as I can tell, `query` will only ever contain one file name and nothing else
+        const getMatches = async (
+            repositories: string[],
+            query: string,
+            count = 30
+        ): Promise<FuzzyFindFile[] | Error> => {
+            return this.fetchSourcegraphAPI<APIResponse<FuzzyFindFilesResponse>>(FUZZY_FILES_QUERY, {
+                query: `type:path count:${count} ${
+                    repositories.length > 0 ? `repo:^(${repositories.map(escapeRegExp).join('|')})$` : ''
+                } ${query}`,
+            }).then(response =>
+                extractDataOrError(
+                    response,
+                    data => data.search?.results.results ?? new Error('no files found')
+                )
             )
-        )
+        }
+
+        // Expand the path into parts combined with the OR operator
+        // so as to return as many possible results as possible.
+        // Experimentally, simply separating the path parts performed
+        // as well as expaning using regular expressions and wildcard,
+        // and didn't take as long.
+        const expandQuery = (query: string): string => {
+            const { dir, name, ext } = path.parse(query)
+            const parts = [dir !== '.' ? dir.replace('/', ' OR ') : '', name, ext].filter(Boolean)
+
+            return `(${parts.join(' OR ')})`.replace(/\./g, '\\.')
+        }
+
+        const expandedQuery = expandQuery(query)
+
+        // Query ten times as many results as we want to display
+        // so that fuzzysort will have something to work with.
+        // Not sure whhy 30 is the magic number to display.
```

Comment:
```suggestion
        // Not sure why 30 is the magic number to display.
```

Again, find out and comment definitively, or omit this and let someone else find out if they care. But I don't think this comment is pulling its weight.

If you have a stylistic preference about coding or documentation we're failing here get on Slack and let's talk about it as a team.